src/plugins/data/public/ui/query_editor/query_editor_top_row.tsx

```diff
@@ -225,7 +225,17 @@ export default function QueryEditorTopRow(props: QueryEditorTopRowProps) {
   }
 
   function shouldRenderDatePicker(): boolean {
-    return Boolean(props.showDatePicker ?? true) ?? (props.showAutoRefreshOnly && true);
+    return (
+      Boolean((props.showDatePicker || props.showAutoRefreshOnly) ?? true) &&
```

Comment:
thanks! good catch. i think this wouldn't have been caught until a user complained about it